turnpike/plugins/oidc/oidc.py

```diff
@@ -0,0 +1,197 @@
+
+            if not oidc_response.ok:
+                raise UnableCreateKeysetError(
+                    f"Unexpected status code received when fetching the OIDC configuration: {oidc_response.status_code}. Response body: {oidc_response.content.decode()}"
+                )
+
+            self.app.logger.debug('OIDC configuration fetched from "%s"', self.oidc_configuration_url)
+
+            try:
+                jwks_uri = oidc_response.json()["jwks_uri"]
+            except KeyError:
+                raise UnableCreateKeysetError(
+                    f"Unable to decode the JWKs' URI from the OIDC response: {oidc_response.content.decode()}"
+                )
+
+            try:
+                jwks_certificates_response: Response = requests.get(url=jwks_uri)
+            except Exception as e:
+                raise UnableCreateKeysetError(f"Unable to fetch the JWKS certificates from OIDC: {e}")
+
+            if not jwks_certificates_response.ok:
+                raise UnableCreateKeysetError(
+                    f"Unexpected status code received when fetching JWKS certificates: {oidc_response.status_code}. Response body: {oidc_response.content.decode()}"
+                )
+
+            jwks_certificates = jwks_certificates_response.json()
+
+            # Store the retrieved certificates for 24 hours, which is the recommended caching setting for the JWKS
+            # certificates.
+            cache.set(key="oidc_jwks_response", value=jwks_certificates, timeout=86400)
+
+        try:
+            return KeySet.import_key_set(jwks_certificates)
+        except Exception as e:
+            raise UnableCreateKeysetError(f"Unable to create a keyset from the JWKS certificates: {e}")
+
+    def process(self, context, backend_auth: Dict[str, Dict[str, list[Dict[str, Union[str, list[str]]]]]]):
+        if not "oidc" in backend_auth:
+            self.app.logger.error(
+                f'A "JWT" back end was matched, but the back end does not have a "JWT" section defined: {backend_auth}'
+            )
+
+            context.status_code = HTTPStatus.INTERNAL_SERVER_ERROR
+            return context
+
+        bearer_token: Optional[str] = request.headers.get("Authorization")
+        if not bearer_token:
+            self.app.logger.debug('Unauthorized the request because the "Authorization" header is missing')
+
+            context.status_code = http.HTTPStatus.UNAUTHORIZED
+            return context
+
+        # Get the key set which we will use to verify the signature of the certificate.
+        try:
+            key_set: KeySet = self._get_jwks_keyset()
+        except UnableCreateKeysetError as e:
+            self.app.logger.error(f"Unable to generate the keyset to verify the incoming token: {e}")
+
+            context.status_code = http.HTTPStatus.INTERNAL_SERVER_ERROR
+            return context
+
+        # Strip the "Bearer" part of the token if it comes with it.
+        if bearer_token.startswith("Bearer"):
```

Comment:
this is only nitpick 😇 
I am wondering if there can be a case when the `Bearer` substring can be case sensitive + the usage of regex feels me a heavy a little bit ... but I don't have strong preferences here ❤️ ... for example [removeprefix()](https://docs.python.org/3/library/stdtypes.html#str.removeprefix) method can be used here